File: apps/ppp/src/routes/(app)/problems/[problem]/design-patterns/factory/payment-system/runtimes/zig/code.zig
const std = @import("std");

const PaymentSystemType = enum { paypal, webmoney, cat_bank };

fn payment(tp: PaymentSystemType, base: i64, amount: i64) i64 {
    _ = tp;
    _ = base;
    _ = amount;
    @panic("Not implemented");
}

File: apps/ppp/src/routes/(app)/problems/[problem]/design-patterns/factory/payment-system/runtimes/cpp/code.cpp
#include <cstdlib>

enum class SystemType {
  PayPal,
  WebMoney,
  CatBank
};

class Solution {
public:
  static int payment(SystemType type, int base, int amount) {
    (void)type;
    (void)base;
    (void)amount;
    std::abort(); // Not implemented
  }
};
